corehq/apps/domain/models.py

```diff
@@ -1040,6 +1040,14 @@ def location_types(self):
         return LocationType.objects.filter(domain=self.name).all()
 
     @property
+    @memoized
+    def uses_locations(self):
+        if self.commtrack_enabled:
+            return True
+        from corehq.apps.locations.models import LocationType
+        return LocationType.objects.filter(domain=self.name).exists()
```

Comment:
I expect this will replace the `locations_enabled` flag once the feature preview is removed.  We still want to hide certain features if the domain doesn't actually use locations.  Sheel suggested having defined location types as the indicator for whether or not the project cares about locations.